### PR TITLE
Add cap to use idevicescreenshot for real device screenshots

### PR DESCRIPTION
**Pull request by imurchie**
The flow for screenshotting on real devices as it currently stands:
1. Try WDA method (i.e., `GET` request to `/screenshot` on WDA)
2. Try `idevicescreenshot` if available
3. Try WDA method an additional two times

However, there are reported cases where WDA still crashes when getting screenshots

This PR
* Adds `realDeviceScreenshotter` cap to specify the program to use preferentially (for now, either nothing, to use WDA first, or `idevicescreenshot` to use the `libimobiledevice` program). In the former case, the flow remains as above. In the case of using `idevicescreenshot` the flow becomes
    1. Try `idevicescreenshot` if available
* Fixes handling of existence of `idevicescreenshot` (`fs.which` throws an error when program not found)

Resolves https://github.com/appium/appium/issues/10590

### CODE REVIEW

File: lib/commands/screenshots.js [hunk 1]
```diff
@@ -42,51 +42,62 @@ async function getScreenshotWithIdevicelib (udid, isLandscape) {
...
   const getScreenshotFromWDA = async () => {
+    log.debug(`Taking screenshot with WDA`);
     const data = await this.proxyCommand('/screenshot', 'GET');
     if (!_.isString(data)) {
       throw new Error(`Unable to take screenshot. WDA returned '${JSON.stringify(data)}'`);
     }
     return data;
   };
 
+  const useIdeviceScreenshot = this.opts.realDeviceScreenshotter &&
```
Comment (mykola-mokhnach) [inline lib/commands/screenshots.js]: you could also use `_.lowerCase(this.opts.realDeviceScreenshotter) === 'idevicescreenshot'`

File: lib/commands/screenshots.js [hunk 2]
```diff
@@ -42,51 +42,62 @@ async function getScreenshotWithIdevicelib (udid, isLandscape) {
...
+  try {
+    await fs.which('idevicescreenshot');
+  } catch (err) {
+    throw new Error(`No 'idevicescreenshot' program found. To use, install ` +
+                    `using 'brew install --HEAD libimobiledevice'`);
+  }
 }
 
 commands.getScreenshot = async function () {
+  const getScreenshotFromLIMD = async () => {
```
Comment (mykola-mokhnach) [inline lib/commands/screenshots.js]: What is LIMD?

**Comment by mykola-mokhnach** [comment]
It looks like unit tests still need some fixes

File: lib/commands/screenshots.js (hunk 2 above)
Comment (jlipps) [reply lib/commands/screenshots.js]: libimobiledevice

File: lib/commands/screenshots.js [hunk 3]
```diff
@@ -42,51 +42,62 @@ async function getScreenshotWithIdevicelib (udid, isLandscape) {
 }
 
 async function isIdevicescreenshotAvailable () {
```
[the remark this answers is not in the document]
Comment (jlipps) [reply lib/commands/screenshots.js]: good point, maybe the name should be `verifyIdeviceScreenshotAvailable` if it's making an assertion instead

File: lib/commands/screenshots.js [hunk 4]
```diff
@@ -41,52 +41,62 @@ async function getScreenshotWithIdevicelib (udid, isLandscape) {
...
     const data = await this.proxyCommand('/screenshot', 'GET');
     if (!_.isString(data)) {
       throw new Error(`Unable to take screenshot. WDA returned '${JSON.stringify(data)}'`);
     }
     return data;
   };
 
+  const useIdeviceScreenshot = _.lowerCase(this.opts.realDeviceScreenshotter) === 'idevicescreenshot';
+  if (useIdeviceScreenshot) {
+    return await getScreenshotFromIDS();
```
Comment (mykola-mokhnach) [inline lib/commands/screenshots.js]: So, we don't even try to get the screenshot with WDA if this fails and the capability is set, right?

Comment (imurchie) [reply lib/commands/screenshots.js]: correct. In the cases in question, WDA crashes and nothing further can be done, in these cases.

Comment (mykola-mokhnach) [reply lib/commands/screenshots.js]: it would be interesting to figure out when exactly it crashes and where.

Comment (mykola-mokhnach) [reply lib/commands/screenshots.js]: perhaps something can still be done there, at least not to crash it totally

Comment (imurchie) [reply lib/commands/screenshots.js]: I can't reproduce, and don't have access to the app that is causing the problems.

**Pull request merged**